src/components/modals/AddEditModal.js

```diff
@@ -0,0 +1,195 @@
+import React, { forwardRef, useMemo, useState } from 'react';
+import ReactModal from 'react-modal';
+import { useModal } from 'react-modal-hook';
+import Select from 'react-select';
+import ReactDatePicker from 'react-datepicker';
+import styled from 'styled-components';
+import {
+  modalStyles,
+  PrimaryButtonOutlined,
+  PrimaryButtonFilled,
+  CloseIcon,
+  ModalActions,
+} from '../../shared/modalstyles';
+
+const LeftContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-basis: 60%;
+  gap: 13px;
+`;
+
+const TopContainer = styled.div`
+  margin-top: 30px;
+  display: flex;
+  flex: 1;
+  flex-direction: row;
+  gap: 30px;
+`;
+
+const RightContainer = styled.div`
+  display: flex;
+  flex-basis: 40%;
+  flex-direction: column;
+  gap: 13px;
+`;
+
+const BottomContainer = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  margin-top: 30px;
+  margin-bottom: 60px;
+`;
+
+const FieldLabel = styled.label`
+  margin-bottom: 13px;
+  color: 'red';
+`;
+
+const Datepicker = styled.div`
+  box-sizing: border-box;
+  border-radius: 0;
+  background-color: darkgrey;
+  border: none;
+  outline: none;
+  height: 57px;
+  padding: 18px;
+  color: white;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const StyledInput = styled.input`
+  box-sizing: border-box;
+  border-radius: 0;
+  background-color: darkgrey;
+  border: none;
+  outline: none;
+  height: 57px;
+  padding-left: 18px;
+  color: white;
+`;
+
+const StyledTextArea = styled.textarea`
+  border-radius: 0;
+  background-color: darkgrey;
+  border: none;
+  outline: none;
+  height: 57px;
+  padding: 18px;
+  color: white;
+  resize: none;
+`;
+
+const customStyles = {
+  control: (styles) => ({
+    ...styles,
+    backgroundColor: 'darkgrey',
+    border: 'none',
+    borderRadius: 0,
+    height: '57px',
+    boxShadow: 'none',
+    paddingLeft: '18px',
+  }),
+  input: (styles) => ({
+    ...styles,
+    color: 'white',
+    padding: 0,
+  }),
+  option: (styles) => ({
+    ...styles,
+    color: 'grey',
+  }),
+};
+
+const UseAddEditMovieModal = (flowType) => {
+  const [selectedMovieGenre, setSelectedMovieGenre] = useState({ name: 'comedy' });
+  const [releaseDate, setReleaseDate] = useState(new Date());
+
+  const CustomDatePickerInput = forwardRef((props, ref) => {
+    return (
+      <Datepicker onClick={props.onClick} ref={ref} role="button">
+        {props.value}
+      </Datepicker>
+    );
+  });
+
+  CustomDatePickerInput.displayName = 'CustomDatePickerInput';
+  const genres = [
+    { name: 'documentary' },
+    { name: 'comedy' },
+    { name: 'horror' },
+    { name: 'crime' },
+  ];
+  const movieGenres = useMemo(() => {
+    return genres.map((item) => {
+      return { value: item.name, label: item.name };
+    });
+  }, []);
```

Comment:
You have missed added the dependency here. Checkout on the eslint error from your code editor.

I see the main idea here is to generate a data structure that supports a drop down or select from an array of objects like genre above. Could you consider moving this also to some re-usable utility and re-use it throughout the application on need basis? This is how any real time application is designed.